packages/server/src/integration-test/mysql.spec.ts

```diff
@@ -0,0 +1,1289 @@
+      foreignRows.push({
+        row: foreignRow,
+        relationshipType: RelationshipType.MANY_TO_ONE,
+      })
+    }
+
+    for (let i = 0; i < (opts?.createForeignRows?.createManyToMany || 0); i++) {
+      const foreignRow = await config.createRow({
+        tableId: manyToManyRelationshipInfo.table._id,
+        title: generator.name(),
+      })
+
+      rowData = {
+        ...rowData,
+        [manyToManyRelationshipInfo.fieldName]:
+          rowData[manyToManyRelationshipInfo.fieldName] || [],
+      }
+      rowData[manyToManyRelationshipInfo.fieldName].push(foreignRow._id)
+      foreignRows.push({
+        row: foreignRow,
+        relationshipType: RelationshipType.MANY_TO_MANY,
+      })
+    }
+
+    const row = await config.createRow({
+      tableId: primaryMySqlTable._id,
+      ...rowData,
+    })
+
+    return { row, foreignRows }
+  }
+
+  async function createDefaultMySqlTable() {
+    return await config.createTable({
+      name: generator.word({ length: 10 }),
+      type: "table",
+      primary: ["id"],
+      schema: {
+        id: {
+          name: "id",
+          type: FieldType.AUTO,
+          autocolumn: true,
+        },
+      },
+      sourceId: mysqlDatasource._id,
+      sourceType: TableSourceType.EXTERNAL,
+    })
+  }
+
+  const createRandomTableWithRows = async () => {
+    const tableId = (await createDefaultMySqlTable())._id!
+    return await config.api.row.save(tableId, {
+      tableId,
+      title: generator.name(),
+    })
+  }
+
+  async function populatePrimaryRows(
+    count: number,
+    opts?: {
+      createOneToMany?: boolean
+      createManyToOne?: number
+      createManyToMany?: number
+    }
+  ) {
+    return await Promise.all(
+      Array(count)
+        .fill({})
+        .map(async () => {
+          const rowData = generateRandomPrimaryRowData()
+          return {
+            rowData,
+            ...(await createPrimaryRow({
+              rowData,
+              createForeignRows: opts,
+            })),
+          }
+        })
+    )
+  }
+
+  it("validate table schema", async () => {
+    const res = await makeRequest(
+      "get",
+      `/api/datasources/${mysqlDatasource._id}`
+    )
+
+    expect(res.status).toBe(200)
+    expect(res.body).toEqual({
+      config: {
+        database: "mysql",
+        host: mysqlDatasource.config!.host,
+        password: "--secret-value--",
+        port: mysqlDatasource.config!.port,
+        user: "root",
+      },
+      plus: true,
+      source: "MYSQL",
+      type: "datasource_plus",
+      _id: expect.any(String),
+      _rev: expect.any(String),
+      createdAt: expect.any(String),
+      updatedAt: expect.any(String),
+      entities: expect.any(Object),
+    })
+  })
+
+  describe("POST /api/:tableId/rows", () => {
+    const createRow = (tableId: string | undefined, body: object) =>
+      makeRequest("post", `/api/${tableId}/rows`, body)
```

Comment:
You have access to `config` in these tests, so you should also have access to `config.api.row.create` and such. I know you didn't write this initially, and you're copying what's in `postgres.spec.ts`, but the stuff in there is quite old and crufty compared to how we write tests today. It might not be the best idea to copy it. Going to tag @mike12345567 to get his opinion.